ui-tests/src/test/java/plugins/WarningsNextGenerationPluginTest.java

```diff
@@ -0,0 +1,621 @@
+        return new AnalysisSummary(build, toolId).openInfoView();
+    }
+
+    /**
+     * Tests the build overview page by running two builds that aggregate the three different tools into a single
+     * result. Checks the contents of the result summary.
+     */
+    @Test
+    public void should_aggregate_tools_into_single_result() {
+        FreeStyleJob job = createFreeStyleJob("build_status_test/build_01");
+        IssuesRecorder recorder = addRecorder(job);
+        recorder.setEnabledForAggregation(true);
+        recorder.addQualityGateConfiguration(4, QualityGateType.TOTAL, QualityGateBuildResult.UNSTABLE);
+        recorder.addQualityGateConfiguration(3, QualityGateType.NEW, QualityGateBuildResult.FAILED);
+        recorder.setIgnoreQualityGate(true);
+
+        job.save();
+
+        Build referenceBuild = buildJob(job).shouldBeUnstable();
+        referenceBuild.open();
+
+        assertThat(new AnalysisSummary(referenceBuild, CHECKSTYLE_ID)).isNotDisplayed();
+        assertThat(new AnalysisSummary(referenceBuild, PMD_ID)).isNotDisplayed();
+        assertThat(new AnalysisSummary(referenceBuild, FINDBUGS_ID)).isNotDisplayed();
+
+        AnalysisSummary referenceSummary = new AnalysisSummary(referenceBuild, ANALYSIS_ID);
+        assertThat(referenceSummary).isDisplayed()
+                .hasTitleText("Static Analysis: 4 warnings")
+                .hasAggregation("FindBugs, CPD, CheckStyle, PMD")
+                .hasNewSize(0)
+                .hasFixedSize(0)
+                .hasReferenceBuild(0);
+
+        reconfigureJobWithResource(job, "build_status_test/build_02");
+
+        Build build = buildJob(job);
+
+        build.open();
+
+        AnalysisSummary analysisSummary = new AnalysisSummary(build, ANALYSIS_ID);
+        assertThat(analysisSummary).isDisplayed()
+                .hasTitleText("Static Analysis: 25 warnings")
+                .hasAggregation("FindBugs, CPD, CheckStyle, PMD")
+                .hasNewSize(23)
+                .hasFixedSize(2)
+                .hasReferenceBuild(1);
+
+        AnalysisResult result = analysisSummary.openOverallResult();
+        assertThat(result).hasActiveTab(Tab.TOOLS).hasTotal(25)
+                .hasOnlyAvailableTabs(Tab.TOOLS, Tab.PACKAGES, Tab.FILES, Tab.CATEGORIES, Tab.TYPES, Tab.ISSUES);
+    }
+
+    private IssuesRecorder addRecorder(final FreeStyleJob job) {
+        return job.addPublisher(IssuesRecorder.class, recorder -> {
+            recorder.setTool("CheckStyle");
+            recorder.addTool("FindBugs");
+            recorder.addTool("PMD");
+            recorder.addTool("CPD",
+                    cpd -> cpd.setHighThreshold(8).setNormalThreshold(3));
+            recorder.setEnabledForFailure(true);
+        });
+    }
+
+    /**
+     * Test to check that the issue filter can be configured and is applied.
+     */
+    @Test
+    public void should_filter_issues_by_include_and_exclude_filters() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'should_filter_issues_by_include_and_exclude_filters' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5070362)